rpcadaptors.go

```diff
@@ -324,3 +328,405 @@ func (b *rpcSyncMgr) ProcessTransaction(tx *dcrutil.Tx, allowOrphans bool,
 	return b.blockMgr.ProcessTransaction(tx, allowOrphans,
 		rateLimit, allowHighFees, tag)
 }
+
+// rpcUtxoEntry represents a utxo entry for use with the RPC server and
+// implements the rpcserver.UtxoEntry interface.
+type rpcUtxoEntry struct {
+	utxoEntry *blockchain.UtxoEntry
```

Comment:
Since the interface essentially is already implemented by `blockchain.UtxoEntry`, you can avoid almost all of this code in here by simply embedding the type directly.  Then you'd only need to define the `ToUtxoEntry()` case since all the others would automatically be inherited.

e.g. The following will implement it properly:

```Go
// rpcUtxoEntry represents a utxo entry for use with the RPC server and
// implements the rpcserver.UtxoEntry interface.
type rpcUtxoEntry struct {
	*blockchain.UtxoEntry
}

// Ensure rpcUtxoEntry implements the rpcserver.UtxoEntry interface.
var _ rpcserver.UtxoEntry = (*rpcUtxoEntry)(nil)

// ToUtxoEntry returns the underlying UtxoEntry instance.
func (u *rpcUtxoEntry) ToUtxoEntry() *blockchain.UtxoEntry {
	return u.UtxoEntry
}
```

Due to this, I'm skipping reviewing all of the other unnecessary funcs in this file since they'll be removed.